src/algebra/star/basic.lean

```diff
@@ -266,18 +277,32 @@ def star_ring_of_comm {R : Type*} [comm_semiring R] : star_ring R :=
   ..star_monoid_of_comm }
 
 /--
-An ordered `*`-ring is a ring which is both an ordered ring and a `*`-ring,
-and `0 ≤ star r * r` for every `r`.
-
-(In a Banach algebra, the natural ordering is given by the positive cone
-which is the closure of the sums of elements `star r * r`.
-This ordering makes the Banach algebra an ordered `*`-ring.)
+An ordered `*`-ring is a ring which is both an `ordered_add_comm_group` and a `*`-ring,
+and `0 ≤ r ↔ ∃ s, r = star s * s`.
 -/
-class star_ordered_ring (R : Type u) [ordered_semiring R] extends star_ring R :=
-(star_mul_self_nonneg : ∀ r : R, 0 ≤ star r * r)
+class star_ordered_ring (R : Type u) [ring R] [partial_order R] extends star_ring R :=
+(add_le_add_left       : ∀ a b : R, a ≤ b → ∀ c : R, c + a ≤ c + b)
```

Comment:
You can also obtain `add_le_add_left` via `covariant_class R R (+) (≤)`.  Indeed, after your definition, this works:
```lean
instance : covariant_class R R (+) (≤) := ⟨λ a b c bc, add_le_add_left b c bc a⟩
```

Reply:
I just looked into this, and at the moment there is a `[covariant_class R R (+) (≤)]` instance given `[ordered_add_comm_group R]` and not the other way around, so for now I think it's better to leave it as is. Maybe it would make sense to redefine `ordered_add_comm_group` in terms of `covariant_class`, but I think I'd rather leave that job to someone with a better overall view of the order hierarchy.

Reply:
> there is a `[covariant_class R R (+) (≤)]` instance given `[ordered_add_comm_group R]` and not the other way around

Right, this is the same direction as what @adomani is suggesting. I think adding the snippet in the comment above is a good idea.

Reply:
This is also an argument for reordering the arguments to match:
```suggestion
(add_le_add_left       : ∀ a b c : R, a ≤ b → c + a ≤ c + b)
```

Reply:
> Right, this is the same direction as what @adomani is suggesting. I think adding the snippet in the comment above is a good idea.

But there is already an `[ordered_add_comm_group R]` instance right below this, which implies `[covariant_class R R (+) (≤)]`, so I'm not sure why we would want both.

Reply:
Ah, I hadn't spotted that, then this is fine.